File: JAVA/3066. Minimum Operations to Exceed Threshold Value II.java
class Solution {
    public int minOperations(int[] n, int k) {
        PriorityQueue<Long>p=new PriorityQueue<>();
        for(int i:n){
            p.add((long)i);
        }
        int c=0;
        while(p.size()>1){
            long x=p.poll();
            if(x>=k)
            return c;
            long y=p.poll();
            p.add(Math.min(x,y)*2+Math.max(x,y));
            c++;
        }
        return c;
    }
}

File: C++/3066. Minimum Operations to Exceed Threshold Value II.cpp
class Solution {
public:
    int minOperations(vector<int>& n, int k) {
        priority_queue<long,vector<long>,greater<long>>p(n.begin(),n.end());
        int c=0;
        while(p.size()>1){
            if(p.top()>=k)
            return c;
            long x=p.top();
            p.pop();
            long y=p.top();
            p.pop();
            p.push(x*2+y);
            c++;
        }
        return c;
    }
};   